### PR TITLE
TST : Added test for creating empty dataframe with column of type str…

### LINKED ISSUE #34915: BUG: Cannot create empty dataframe having columns with dtype string
- [x] I have checked that this issue has not already been reported.

I have checked, but this requirement is grossly unreasonable since there are currently 3,426 other open issues. It's impossible for anyone to be sure. This requirement should be reworded.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python-traceback
>>> import pandas as pd

>>> pd.DataFrame(columns=['c1'], dtype='string')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/homeuser/PycharmProjects/.venv/myenv/lib/python3.8/site-packages/pandas/core/frame.py", line 435, in __init__
    mgr = init_dict(data, index, columns, dtype=dtype)
  File "/home/homeuser/PycharmProjects/.venv/myenv/lib/python3.8/site-packages/pandas/core/internals/construction.py", line 234, in init_dict
    if dtype is None or np.issubdtype(dtype, np.flexible):
  File "/home/homeuser/PycharmProjects/.venv/myenv/lib/python3.8/site-packages/numpy/core/numerictypes.py", line 388, in issubdtype
    arg1 = dtype(arg1).type
TypeError: Cannot interpret 'StringDtype' as a data type
```

#### Problem description
These three alternatives work, so there is no reason for this bug to continue to not be fixed:
```python-traceback
>>> pd.DataFrame(columns=['c1'], dtype='str')
Empty DataFrame
Columns: [c1]
Index: []

>>> pd.DataFrame([{}], columns=['c1'], dtype='string')
     c1
0  <NA>

>>> pd.DataFrame([{}], columns=['c1'], dtype='string').drop(labels=0)
Empty DataFrame
Columns: [c1]
Index: []
```

#### Expected Output
```python-traceback
Empty DataFrame
Columns: [c1]
Index: []
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-1087-oem
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 1.0.5
numpy            : 1.19.0
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 40.8.0
Cython           : None
pytest           : 5.4.3
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.5.1
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.5.1
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 5.4.3
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by prakhar987**
- [x] closes #34915
- [x] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by prakhar987** [comment]
For now I have placed the test in ```tests/frame/test_constructors.py``` but maybe we don't want tests for extension dtypes in that file, in which case i can move it elsewhere.

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -1643,6 +1643,11 @@ def test_constructor_empty_with_string_dtype(self):
         df = DataFrame(index=[0, 1], columns=[0, 1], dtype="U5")
         tm.assert_frame_equal(df, expected)
 
+        # GH 34915
```
Comment (mroeschke) [inline pandas/tests/frame/test_constructors.py:1646]: Could you create a separate test function?

Comment (prakhar987) [reply pandas/tests/frame/test_constructors.py:1647]: Done

**Review by mroeschke — APPROVED** [review]

**Comment by WillAyd** [comment]
lgtm - @TomAugspurger any thoughts on location of test before merge?

**Comment by TomAugspurger** [comment]
This looks good, thanks.

**Pull request merged**